F-Clip Docker Image

kirill-ivanov-a opened the pull request:
Solved tasks:
- [x] F-Clip adapter code added;
- [x] Docker Image created.

kirill-ivanov-a committed:
F-Clip adapter added

- F-Clip adapted;
- Docker Image created.

kirill-ivanov-a committed:
F-Clip adapter added

- F-Clip adapted;
- Docker Image created.

kirill-ivanov-a committed:
Typo corrected

kirill-ivanov-a committed:
Typo corrected

kirill-ivanov-a committed:
Merge branch 'f-clip' of github.com:prime-slam/line-detection-dockers into f-clip

kirill-ivanov-a committed:
CPU supported

- CPU support added;
- Hardcoded paths removed
- README updated;
- minor cleanup.

kirill-ivanov-a committed:
Minor cleanup

kirill-ivanov-a committed:
Update README.md

anastasiia-kornilova commented on F-Clip/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+FROM pytorch/pytorch:1.11.0-cuda11.3-cudnn8-runtime
```
Please, add everywhere license, with you, me and Dmitrii as authors. I suppose Apache 2.0 will be enough

anastasiia-kornilova commented on F-Clip/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+FROM pytorch/pytorch:1.11.0-cuda11.3-cudnn8-runtime
+WORKDIR /app
```
Don't like this name for a folder that contains algorithm

anastasiia-kornilova replied:
I would still make it common for all dockers for easier automation, so let's simply rename

anastasiia-kornilova commented on F-Clip/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+# download pretrained models
+RUN python3 -m pip install --no-cache-dir gdown  \
+    && gdown "1jMZB-kYTNAGaVPW0FBLTAQMc3m3vR-_P" \
+    && apt-get install -y --no-install-recommends unzip \
```
Let's have all dependencies in one step to clearly see what tools do we need, so I would move it up

anastasiia-kornilova commented on F-Clip/FClip/line_dataset.py:
```diff
@@ -0,0 +1,40 @@
+import glob
```
Let's have nice sorting of imports (first -- external libs `import`, then external `from`, then local imports) and separation lines between

anastasiia-kornilova commented on F-Clip/FClip/line_dataset.py:
```diff
@@ -0,0 +1,40 @@
+
+class LineDataset(Dataset):
+    def __init__(self, data_path, image_rescale=(512, 512)):
+        files = glob.glob(f"{data_path}/*")
```
I would prefer using `Path` lib, it has nicer option for listing dirs and also easier usage for basename extraction of files, you can check nice arguments here -- https://habr.com/ru/post/453862/

anastasiia-kornilova commented on F-Clip/FClip/line_dataset.py:
```diff
@@ -0,0 +1,40 @@
+
+    def __getitem__(self, idx):
+        path = self.files[idx]
+        image_name = path.split("/")[-1].split(".")[0]
```
Should be simplified with `Path`

anastasiia-kornilova commented on F-Clip/FClip/line_dataset.py:
```diff
@@ -0,0 +1,40 @@
+
+        meta = {"width": width, "height": height, "image_name": image_name}
+
+        image = (image_ - M.image.mean) / M.image.stddev
```
Please, add comment here on what is going on, not clear

kirill-ivanov-a committed:
Minor fixes

- License added;
- string paths replaced with pathlib;
- docker dependencies groupped;
- imports sorted.

kirill-ivanov-a committed:
Merge branch 'f-clip' of github.com:prime-slam/line-detection-dockers into f-clip

kirill-ivanov-a committed:
License updated

anastasiia-kornilova commented on F-Clip/Dockerfile:
```diff
@@ -0,0 +1,38 @@
+    libgl1 \
+    libglib2.0-0 \
+    unzip \
+    && python3 -m pip install --no-cache-dir gdown -r requirements.txt
```
Not sure about this idea, how docker cache will behave if requirements will be updated?

kirill-ivanov-a committed:
Docker dependency layer separated
